src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift

```diff
@@ -0,0 +1,198 @@
+		descriptionLabelResetNotifications.font = UIFont.enaFont(for: .subheadline)
+		
+		let resetNotificationsButton = UIButton(frame: .zero)
+		resetNotificationsButton.translatesAutoresizingMaskIntoConstraints = false
+		resetNotificationsButton.setTitle("Reset notification state", for: .normal)
+		resetNotificationsButton.addTarget(self, action: #selector(resetNotificationsButtonTapped), for: .touchUpInside)
+		resetNotificationsButton.setTitleColor(.enaColor(for: .buttonDestructive), for: .normal)
+		
+		time1Label = UILabel(frame: .zero)
+		time1Label.translatesAutoresizingMaskIntoConstraints = false
+		time1Label.numberOfLines = 0
+		time1Label.text = "Notification 1 time (sec)"
+		
+		time2Label = UILabel(frame: .zero)
+		time2Label.translatesAutoresizingMaskIntoConstraints = false
+		time2Label.numberOfLines = 0
+		time2Label.text = "Notification 2 time (sec)"
+		
+		time1Textfield = UITextField(frame: .zero)
+		time1Textfield.translatesAutoresizingMaskIntoConstraints = false
+		time1Textfield.delegate = self
+		time1Textfield.borderStyle = .line
+		
+		time2Textfield = UITextField(frame: .zero)
+		time2Textfield.translatesAutoresizingMaskIntoConstraints = false
+		time2Textfield.delegate = self
+		time2Textfield.borderStyle = .line
+	
+		
+		let stackView = UIStackView(arrangedSubviews: [titleLabel, descriptionLabel, time1Label, time1Textfield, time2Label, time2Textfield, saveButton, resetDefaultsButton, descriptionLabelResetNotifications, resetNotificationsButton])
+		stackView.translatesAutoresizingMaskIntoConstraints = false
+		stackView.axis = .vertical
+		stackView.alignment = .center
+		stackView.spacing = 10
+		stackView.setCustomSpacing(20, after: descriptionLabel)
+		
+		view.addSubview(stackView)
+		NSLayoutConstraint.activate([
+			stackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 10),
+			stackView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -10),
+			stackView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 10),
+			time1Textfield.widthAnchor.constraint(equalToConstant: 60),
+			time2Textfield.widthAnchor.constraint(equalToConstant: 60)
+		])
+		
+		// MARK: Set Default Value for the notification text filelds:
+		time1Textfield.text = "\(warnOthers.notificationOneTimer)"
+		time2Textfield.text = "\(warnOthers.notificationTwoTimer)"
+		
+		//Set the keyboard type.
+		time1Textfield.keyboardType = .numberPad
+		time2Textfield.keyboardType = .numberPad
+		self.hideKeyboardWhenTappedAround()
+		
+	}
+	
+	// MARK: Private API
+	@objc
+	private func scheduleNotificationsButtonTapped() {
+		let time1 = Int(time1Textfield.text ?? "")
+		let time2 = Int(time2Textfield.text ?? "")
+		
+		// Create an alert when user enter the wrong values.
+		if time2 ?? WarnOthersNotificationsTimer.timerOneTime.rawValue <= time1 ?? WarnOthersNotificationsTimer.timerTwoTime.rawValue {
```

Comment:
This is hard to read, you could improve this by putting the `?? WarnOthersNotificationsTimer.timerOneTime.rawValue` at the end of line 139. And the other one accordingly

Reply:
Refactored